src/pixel_mapper.rs

```diff
@@ -0,0 +1,43 @@
+use crate::multiplex_mapper::MultiplexMapper;
+
+/// A pixel mapper is a way for you to map pixels of LED matrixes to a different
+/// layout. If you have an implementation of a PixelMapper, you can give it
+/// to the RGBMatrix::apply_pixel_mapper(), which then presents you with a canvas
+/// that has the new "visible width" and "visible height".
+pub(crate) trait PixelMapper {
+    /// Given a underlying matrix (width, height), returns the
+    /// visible (width, height) after the mapping.
+    /// E.g. a 90 degree rotation might map matrix=(64, 32) -> visible=(32, 64)
+    /// Some multiplexing matrices will double the height and half the width.
+    fn get_size_mapping(&self, matrix_width: usize, matrix_height: usize) -> [usize; 2];
+
+    /// Map where a visible pixel (x,y) is mapped to the underlying matrix (x,y).
+    fn map_visible_to_matrix(
+        &self,
+        matrix_width: usize,
+        matrix_height: usize,
+        visible_x: usize,
+        visible_y: usize,
+    ) -> [usize; 2];
+}
+
+pub(crate) struct MultiplexMapperWrapper(pub(crate) Box<dyn MultiplexMapper>);
```

Comment:
That is a lot of redirection, did you try to implement `PixelMapper` for `MultiplexMapper`?

Reply:
We could implement `PixelMapper` for `MultiplexMapper` using Nightly and [trait_upcasting](https://doc.rust-lang.org/beta/unstable-book/language-features/trait-upcasting.html) feature. [This example](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=d628c3356784139b21ceebff23c9f967) demonstrates that.

I proposed wrappers because I didn't want to use an unstable feature.

I am not suggesting using wrappers is the only way to achieve this. There may be a better way. If you have any insights, I would appreciate hearing them.

Reply:
Thanks for looking into it. I don't want to hold you up. Let's go with this solution for now, we can look into how to improve this later.